# HG changeset patch
# Parent 5b539cad5f3de90a11e711e1bc57b19072f7236d
include/llvm/Target/TargetAsmBackend.h has been moved to include/MC/MCAsmBackend.h

diff -r 5b539cad5f3d -r 3aab4cee678d include/llvm/MC/MCAsmBackend.h
--- a/include/llvm/MC/MCAsmBackend.h	Sun Aug 07 14:52:42 2011 -0700
+++ b/include/llvm/MC/MCAsmBackend.h	Sun Aug 07 14:56:06 2011 -0700
@@ -124,6 +124,24 @@
   /// HandleAssemblerFlag - Handle any target-specific assembler flags.
   /// By default, do nothing.
   virtual void HandleAssemblerFlag(MCAssemblerFlag Flag) {}
+
+  // @LOCALMOD-BEGIN
+  /// getBundleSize - Return the size (in bytes) of code bundling units
+  /// for this target. If 0, bundling is disabled. This is used exclusively
+  /// for Native Client.
+  virtual unsigned getBundleSize() const {
+    return 0;
+  }
+
+  /// CustomExpandInst -
+  ///   If the MCInst instruction has a custom expansion, write it to the
+  /// MCStreamer 'Out'. This can be used to perform "last minute" rewrites of
+  /// MCInst instructions for emission.
+  virtual bool CustomExpandInst(const MCInst &Inst, MCStreamer &Out) const {
+    return false;
+  }
+  // @LOCALMOD-END
+
 };
 
 } // End llvm namespace
